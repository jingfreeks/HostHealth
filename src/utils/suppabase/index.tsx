export {default as supabase} from './suppabase';
