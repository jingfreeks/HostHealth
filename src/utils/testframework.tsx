import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
// import {QueryClient, QueryClientProvider} from '@tanstack/react-query';
// // import {DARK_THEME} from '@/utils/constants';
// // import {ThemeProvider} from 'styled-components/native';

type MockProviderType = {
  children?: React.ReactNode;
  store?: object;
};
export const MockProvider = (props: MockProviderType) => {
  const {store, children} = props;
  const mockStore = configureStore([thunk]);
  const mocked = mockStore(store);
  return <Provider store={mocked}>{children}</Provider>;
};

export const testingProps = (reference: string): unknown => ({
  testID: reference,
  accessibilityLabel: reference,
});

// const queryClient = new QueryClient({
//   defaultOptions: {
//     queries: {
//       retry: false,
//     },
//   },
// });
// export const wrapper = ({children}) => (
//   <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
// );
