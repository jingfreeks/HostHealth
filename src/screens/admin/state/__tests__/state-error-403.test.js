

import React from 'react'
import { fireEvent } from '@testing-library/react-native';
import {renderWithProviders} from '@/utils/testframeworknew';
import {State} from '../index';

const deleteState=()=>jest.fn()
jest.mock('@/slice',()=>{
  return{
    useGetStateQuery:()=>({
      isLoading:false,
      isSuccess:false,
      isError:true,
      error:{
        status:403
      },
      data:[]
    }),
    useDeleteStatesMutation:()=>[deleteState,{isLoading:false,isError:false}],
  }
})
describe('State admin Error screen', () => {
  it('Should work as expected to get snapshot', () => {
    const all = renderWithProviders(<State />);
    expect(all.toJSON()).toMatchSnapshot();
  });
});
