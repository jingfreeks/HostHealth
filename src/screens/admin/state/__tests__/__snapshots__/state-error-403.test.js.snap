// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`State admin Error screen Should work as expected to get snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <RCTSafeAreaView
    style={
      [
        {},
      ]
    }
  >
    <View
      alignType="center"
      style={
        [
          {
            "alignItems": "center",
            "alignSelf": "center",
            "flexDirection": "column",
          },
        ]
      }
    >
      <Image
        resizeMode="contain"
        size={150}
        source={
          {
            "testUri": "../../../src/assets/pcitiesempty.png",
          }
        }
        style={
          [
            {
              "height": 294.2647058823529,
              "width": 294.2647058823529,
            },
          ]
        }
      />
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "Avenir",
              "fontSize": 16,
              "fontWeight": "400",
            },
          ]
        }
      >
        Token Expired you must logout and login it again
      </Text>
    </View>
  </RCTSafeAreaView>
</View>
`;
